R/build-reference.R

```diff
@@ -247,6 +247,7 @@ examples_env <- function(pkg, seed = 1014, devel = TRUE, envir = parent.frame())
 #' @rdname build_reference
 build_reference_index <- function(pkg = ".") {
   pkg <- section_init(pkg, depth = 1L)
+  check_dst_path_exists(pkg$dst_path)
   dir_create(path(pkg$dst_path, "reference"))
```

Comment:
```suggestion
  create_subdir(pkg$dst_path, "reference")
```

To make sure I understand correctly:

if a pkgdown site: just create new reference dir, otherwise (if !fs::file_exists(pkg$dst_path, "pkgdown.yml")) call `init_site()` then create the new reference dir?

Reply:
Yeah, that's what I was thinking.

Reply:
Ok. I will think a little bit more. Will let you know when it is ready. I realized that this isn't an issue if you don't have `_pkgdown.yml`. You don't run in these issues with BS 3 for example..